pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+type LanguageName string
+
+var (
+	python  LanguageName = "python"
+	java    LanguageName = "java"
+	unknown LanguageName = ""
+)
+
+type Language struct {
+	Name LanguageName
+}
+
+type languageFromCLI struct {
+	name      LanguageName
+	validator func(exe string) bool
+}
+
+// knownPrefixes maps languages names to their prefix
+var knownPrefixes = map[string]languageFromCLI{
+	"python": {name: python},
+	"java": {name: java, validator: func(exe string) bool {
+		if exe == "javac" || exe == "javac.exe" {
+			return false
+		}
+		return true
+	}},
+}
+
+// exactMatches maps an exact exe name match to a prefix
+var exactMatches = map[string]languageFromCLI{
+	"py": {name: python},
+}
```

Comment:
what's the different from `python` entry in `knownPrefixes`?
meaning, why can't `py` be another entry in `knownPrefixes`?